puffin/src/trace.rs

```diff
@@ -96,13 +96,13 @@ impl<M: Matcher> fmt::Display for Knowledge<M> {
 /// client and server extensions, cipher suits or session ID It also holds the concrete
 /// references to the [`Agent`]s and the underlying streams, which contain the messages
 /// which have need exchanged and are not yet processed by an output step.
-pub struct TraceContext<PB: ProtocolBehavior + 'static> {
+pub struct TraceContext<PB: ProtocolBehavior> {
     /// The knowledge of the attacker
     knowledge: Vec<Knowledge<PB::Matcher>>,
     agents: Vec<Agent<PB>>,
     claims: GlobalClaimList<PB::Claim>,
 
-    put_registry: &'static PutRegistry<PB>,
+    put_registry: PutRegistry<PB>,
```

Comment:
Note to myself: Why is put registry no longer required to be static?

Reply:
We don't store a reference anymore but the actual `PutRegistry` (cloned), which allows us to drop the static lifetime. This is made possible by other changes I made to the way the `PutRegistry` is constructed, in order to support the registration of C PUTs.